### PR TITLE
Added support for proxy configuration

### PR DESCRIPTION
**Pull request by aleab**
I should have covered every WebClient/HttpClient usage in both the Web and Local APIs.
The changes have been tested with a couple of those free anonymous proxies that can be found around, but unfortunately I couldn't test if proxy w/ authentication works.

(Addresses issue #192)

### CODE REVIEW

**Comment by aleab** [comment]
Update: I tested (and also had someone else test) these changes with proxies that require authentication as well and everything seems to work fine!
However, I'm not sure if using a plain-text string for the password is the right thing to do from a security point of view. Do you want me to change it to `SecureString` instead, @JohnnyCrazy?

**Comment by JohnnyCrazy** [comment]
IMO string is good enough. 👍 

Currently testing your changes

File: SpotifyAPI/ProxyConfig.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        public string Password { get; set; }
+
+        /// <summary>
+        /// Whether to bypass the proxy server for local addresses.
+        /// </summary>
+        public bool BypassProxyOnLocal { get; set; }
+
+        public void Set(ProxyConfig proxyConfig)
+        {
+            this.Host = proxyConfig?.Host;
```
Comment (JohnnyCrazy) [inline SpotifyAPI/ProxyConfig.cs]: Please remove redundant `this.`

File: SpotifyAPI/Local/Models/Track.cs [hunk 2]
```diff
@@ -102,32 +109,31 @@ public string GetAlbumArtUrl(AlbumArtSize size)
...
         /// </summary>
         /// <param name="size">AlbumArtSize (160,320,640)</param>
+        /// <param name="proxyConfig">Optional proxy settings</param>
         /// <returns>A Bitmap, which is the albumart</returns>
-        public async Task<Bitmap> GetAlbumArtAsync(AlbumArtSize size)
+        public async Task<Bitmap> GetAlbumArtAsync(AlbumArtSize size, ProxyConfig proxyConfig = null)
         {
-            using (WebClient wc = new WebClient())
+            var data = await GetAlbumArtAsByteArrayAsync(size, proxyConfig).ConfigureAwait(false);
+            using (MemoryStream ms = new MemoryStream(data))
```
Comment (JohnnyCrazy) [inline SpotifyAPI/Local/Models/Track.cs]: Don't we need another `null` check here? data could be null, IDK how memory stream handles it tho

File: SpotifyAPI/Local/SpotifyLocalAPI.cs [hunk 3]
```diff
@@ -63,15 +63,23 @@ public ISynchronizeInvoke SynchronizingObject
 
         public event EventHandler<TrackTimeChangeEventArgs> OnTrackTimeChange;
 
-        public SpotifyLocalAPI(int timerIntervall = 50)
+        public SpotifyLocalAPI(int timerIntervall = 50) : this(null, timerIntervall)
         {
-            _rh = new RemoteHandler(new SpotifyLocalAPIConfig());
+        }
+
+        public SpotifyLocalAPI(ProxyConfig proxyConfig, int timerIntervall = 50)
```
Comment (JohnnyCrazy) [inline SpotifyAPI/Local/SpotifyLocalAPI.cs]: What do you think about adding a field of type `ProxyConfig` to `SpotifyLocalAPIConfig`? Would reduce our constructor overloads here and in `RemoteHandler`

File: SpotifyAPI/ProxyConfig.cs (hunk 1 above)
Comment (aleab) [reply SpotifyAPI/ProxyConfig.cs]: Ops I though I removed them all. My Visual Studio/Resharper/CodeMaid adds them automatically sometimes.

File: SpotifyAPI/Local/SpotifyLocalAPI.cs (hunk 3 above)
Comment (aleab) [reply SpotifyAPI/Local/SpotifyLocalAPI.cs]: Seems like the best thing to do. I'll change it ASAP.

**Comment by JohnnyCrazy** [comment]
Nice, so I guess we are good to go @aleab ? 👍

**Comment by aleab** [comment]
I guess so!

**Comment by JohnnyCrazy** [comment]
Perfect, I will add some examples to the docs and publish a new release ASAP

Thanks for your contribution! 😄

**Comment by JohnnyCrazy** [comment]
BTW, do you mind if I add your Toastify project to the [projects ](https://johnnycrazy.github.io/SpotifyAPI-NET/#projects)list? :) @aleab

**Comment by aleab** [comment]
Of course not! :)

**Pull request merged**